### ISSUE TITLE
Either man page is wrong or `--path` does not work ('Fatal: no snapshot ID specified' while passing `--path`)

### ISSUE THREAD

**Issue by sza-1** [restic/restic#3463]
Output of `restic version`
--------------------------
restic 0.12.0 compiled with go1.16rc1 on linux/amd64

How did you run restic exactly?
-------------------------------
```
restic --password-command 'printf test' --repo /tmp/test ls --path /tmp/a
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Local.

Expected behavior
-----------------
restic should list content of all snapshots for /tmp/a or manual page should be updated.

Actual behavior
---------------
restic says `Fatal: no snapshot ID specified`.

Steps to reproduce the behavior
-------------------------------
Create a test repo. Backup some files. Check `restic snapshots` output and run `restic ls` with `--path`.

```
$ restic --password-command 'printf test' --repo /tmp/test init

$ restic --password-command 'printf test' --repo /tmp/test backup /tmp/a
$ restic --password-command 'printf test' --repo /tmp/test backup /tmp/b /tmp/c
$ restic --password-command 'printf test' --repo /tmp/test backup /tmp/b
$ restic --password-command 'printf test' --repo /tmp/test backup /tmp/c

$ restic --password-command 'printf test' --repo /tmp/test snapshots
repository 02692054 opened successfully, password is correct
ID        Time                 Host        Tags        Paths
-------------------------------------------------------------
1e927435  2021-07-25 00:13:10  host                    /tmp/a

b38757b0  2021-07-25 00:13:13  host                    /tmp/b
                                                       /tmp/c

e05ca706  2021-07-25 00:13:20  host                    /tmp/b

9f3cb05a  2021-07-25 00:13:22  host                    /tmp/c
-------------------------------------------------------------
4 snapshots

$ restic --password-command 'printf test' --repo /tmp/test ls --path /tmp/a
Fatal: no snapshot ID specified
```

Do you have any idea what may have caused this?
-----------------------------------------------
I suspect that `restic-ls` documentation may be wrong and `--path` is useless option for `restic ls`.

Do you have an idea how to solve the issue?
-------------------------------------------
No.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

**Comment by rawtaz** [comment]
I don't really understand what you think is wrong. Can you explain which documentation you think is saying <what> and what part of that is wrong? Here's part of the help in `restic help ls`:

~~~
Usage:
  restic ls [flags] snapshotID [dir...]

Flags:
  -h, --help          help for ls
  -H, --host host     only consider snapshots for this host, when no snapshot ID is given (can be specified multiple times)
  -l, --long          use a long listing format showing size and mode
      --path path     only consider snapshots which include this (absolute) path, when no snapshot ID is given
      --recursive     include files in subfolders of the listed directories
      --tag taglist   only consider snapshots which include this taglist, when no snapshot ID is given (default [])
~~~

As you can see it tells you that you need to specify a snapshot ID. But in the command you ran you are not doing that. Naturally then restic tells you that you are missing a snapshot ID (`Fatal: no snapshot ID specified`).

**So in other words:** What makes you think that `restic --password-command 'printf test' --repo /tmp/test ls --path /tmp/a` is a properly formatted command?

**Comment by crazyh** [comment]
> [Edited] Never mind, I get it now. The `restic help ls` suggests one doesn't have to supply a snapshot ID :) Seems fishy, thanks for the report!

Yeah, it seems that the following check in https://github.com/restic/restic/blob/v0.12.0/cmd/restic/cmd_ls.go#L92 is not necessary: :)

```
	if len(args) == 0 {
		return errors.Fatal("no snapshot ID specified")
	}
```

**Comment by hluup** [comment]
This check probably is necessary, because it seems there's actually no support for leaving it empty. See https://github.com/restic/restic/pull/2905

**Comment by MichaelEischer** [comment]
`ls --path /some/path latest` works, so we shouldn't remove that option. It would be possible to suggest to the user to use `latest` as snapshot ID in case `--path` was specified.

**Comment by sza-1** [comment]
> `ls --path /some/path latest` works, so we shouldn't remove that option. It would be possible to suggest to the user to use `latest` as snapshot ID in case `--path` was specified.

So means that there is no way to filter ALL snapshots by specified path, does not?

**Comment by gum3ng** [comment]
`sam@biggie:~/fun/restic$ ./restic -r ~/mybackup/ ls --path /home/sam/test
-> Fatal: no snapshot ID specified`
Instead of exiting with an error, is it a good idea to set snapshot ID to latest if not specified by user and throw the warning 'Warning: no snapshot ID specified, setting the snapshot ID to latest', and continue the execution instead of exiting?

**Comment by MichaelEischer** [comment]
> Instead of exiting with an error, is it a good idea to set snapshot ID to latest if not specified by user and throw the warning 'Warning: no snapshot ID specified, setting the snapshot ID to latest', and continue the execution instead of exiting?

restic should either do the right thing automatically or tell the user what's wrong with the command line. To avoid ambiguities I'd prefer to just provide a better error message, which suggests to add `latest` to the command line if a path was specified by but no (valid) snapshot id.

> So it means that there is no way to filter ALL snapshots by specified path, does not?

`ls`  currently only handles a single snapshot at a time. If `ls` worked without specifying a snapshot id, then things get ambiguous once a user tries to specify a path. Would `ls --path example latest` then refer to the snapshot `latest` or the folder `latest`  in all snapshots matching the path?
